src/main/resources/static/js/results.js

```diff
@@ -0,0 +1,152 @@
+                .attr("min", "1")
+                .attr("max", "100")
+                .attr("value", "7")
+                .attr("id", `nBin${qID}`);
+
+        const svg = d3.select(`#${qID}`)
+            .append("svg")
+                .attr("width", width + margin.left + margin.right)
+                .attr("height", height + margin.top + margin.bottom)
+            .append("g")
+                .attr("transform",
+                    `translate(${margin.left},${margin.top})`);
+
+        // get the data
+        //id are in the format r#, remove first character
+        d3.json(window.location.origin + `/questions/${qID.substring(1)}`).then( function(data) {
+            // X axis: scale and draw:
+            const x = d3.scaleLinear()
+                .domain([parseInt(data.low), parseInt(data.high)])
+                .range([0, width]);
+            data = data.responses;
+            svg.append("g")
+                .attr("transform", `translate(0,${height})`)
+                .call(d3.axisBottom(x));
+
+            // Y axis: initialization
+            const y = d3.scaleLinear()
+                .range([height, 0]);
+            const yAxis = svg.append("g")
+
+            // A function that builds the graph for a specific value of bin
+            function update(nBin) {
+
+                // set the parameters for the histogram
+                const histogram = d3.histogram()
+                    .value(function(d) { return d; })   // I need to give the vector of value
+                    .domain(x.domain())  // then the domain of the graphic
+                    .thresholds(x.ticks(nBin)); // then the numbers of bins
+
+                // And apply this function to data to get the bins
+                const bins = histogram(data);
+
+                // Y axis: update now that we know the domain
+                y.domain([0, d3.max(bins, function(d) { return d.length; })]);   // d3.hist has to be called before the Y axis obviously
+                yAxis
+                    .transition()
+                    .duration(1000)
+                    .call(d3.axisLeft(y));
+
+                // Join the rect with the bins data
+                const u = svg.selectAll("rect")
+                    .data(bins)
+
+                // Manage the existing bars and eventually the new ones:
+                u
+                    .join("rect") // Add a new rect for each new elements
+                    .transition() // and apply changes to all of them
+                    .duration(1000)
+                    .attr("x", 1)
+                    .attr("transform", function(d) { return `translate(${x(d.x0)}, ${y(d.length)})`})
+                    .attr("width", function(d) { return x(d.x1) - x(d.x0) -1 ; })
+                    .attr("height", function(d) { return height - y(d.length); })
+                    .style("fill", "#69b3a2")
+
+            }
+
+
+            // Initialize with 7 bins
+            update(7)
```

Comment:
Is there any reason for 7 (this is fine if it works)

Reply:
nah, just an arbitrary default value